src/msg/async/AsyncMessenger.cc

```diff
@@ -194,15 +194,23 @@ void Processor::accept()
 	} else if (r == -EMFILE || r == -ENFILE) {
 	  lderr(msgr->cct) << __func__ << " open file descriptions limit reached sd = " << listen_socket.fd()
 			   << " errno " << r << " " << cpp_strerror(r) << dendl;
-	  break;
+	  if (++accept_error_num > msgr->cct->_conf->ms_accept_error_nums) {
+	    lderr(msgr->cct) << "Proccessor accept has encountered enough error numbers, just do ceph_abort()." << dendl;
+	    ceph_abort();
+	  }
+	  continue;
 	} else if (r == -ECONNABORTED) {
 	  ldout(msgr->cct, 0) << __func__ << " it was closed because of rst arrived sd = " << listen_socket.fd()
 			      << " errno " << r << " " << cpp_strerror(r) << dendl;
 	  continue;
 	} else {
 	  lderr(msgr->cct) << __func__ << " no incoming connection?"
 			   << " errno " << r << " " << cpp_strerror(r) << dendl;
-	  break;
+	  if (++accept_error_num > msgr->cct->_conf->ms_accept_error_nums) {
```

Comment:
shall we reset this number if `accept()` returns 0?

Reply:
Yes, it needs to be reset!